src/analysis/locally_convex/strong_topology.lean

```diff
@@ -27,22 +27,25 @@ locally convex, bounded convergence
 
 open_locale topology uniform_convergence
 
-variables {E F : Type*}
+variables {𝕜₁ 𝕜₂ E F : Type*}
 
 namespace continuous_linear_map
 
+variables [add_comm_group E] [topological_space E]
+  [add_comm_group F] [topological_space F] [topological_add_group F]
+
 section general
 
-variables [add_comm_group E] [module ℝ E] [topological_space E]
-  [add_comm_group F] [module ℝ F] [topological_space F] [topological_add_group F]
-  [has_continuous_const_smul ℝ F] [locally_convex_space ℝ F]
+variables [normed_field 𝕜₁] [normed_field 𝕜₂] [module 𝕜₁ E] [module 𝕜₂ F] {σ : 𝕜₁ →+* 𝕜₂}
+variables [module ℝ F] [has_continuous_const_smul ℝ F] [locally_convex_space ℝ F]
+  [smul_comm_class 𝕜₂ ℝ F]
 
 lemma strong_topology.locally_convex_space (𝔖 : set (set E)) (h𝔖₁ : 𝔖.nonempty)
   (h𝔖₂ : directed_on (⊆) 𝔖) :
-  @locally_convex_space ℝ (E →L[ℝ] F) _ _ _ (strong_topology (ring_hom.id ℝ) F 𝔖) :=
+  @locally_convex_space ℝ (E →SL[σ] F) _ _ _ (strong_topology σ F 𝔖) :=
```

Comment:
I was able to generalize `ℝ` to an arbitrary `ordered_semiring R` too.

I've pushed a commit; feel free to revert it if you think it's a bad idea

bors d+

Reply:
Thanks, I was not thinking about that. I think there is no applications for locally convex spaces over ordered semirings that are not the reals, but if the definition of locally convex allows for more generality, then these theorems should as well.

Reply:
Presumably you could work with rational coordinates in R^3, which would make sense here?

Reply:
I didn't get a mail that CI went through. Does
bors merge
work in a comment?

Reply:
Yep, that worked.